src/shared/sam/debugger/awsSamDebugger.ts

```diff
@@ -177,7 +178,16 @@ export class SamDebugConfigProvider implements vscode.DebugConfigurationProvider
                     }
                     const resources = getResourcesFromTemplateDatum(templateDatum)
                     for (const resourceKey of resources.keys()) {
-                        const runtimeName = resources.get(resourceKey)?.Properties?.Runtime
+                        const resource = resources.get(resourceKey)
+                        // Skip it if there is not actually a resource or it's not a funciton
+                        if (
+                            !resource ||
+                            (resource.Type !== CloudFormation.LAMBDA_FUNCTION_TYPE &&
+                                resource.Type !== CloudFormation.SERVERLESS_FUNCTION_TYPE)
```

Comment:
thanks for noticing/handling this. Now I notice that all of the `src/testFixtures/workspaceFolder/**/template.yaml` files only have `AWS::Serverless::Function` resources. Would you mind adding some other type of sample resource in those template files? 

They are used by the tests: https://github.com/aws/aws-toolkit-vscode/blob/76b9f33d7333eedf565c9c1a37d8a9b0ff2b69b1/src/test/shared/sam/debugger/samDebugConfigProvider.test.ts#L909

Hopefully this doesn't require any change to the tests (since you're filtering out such resources now), and having those sample resources in the fixtures serves as a regression guard.

Reply:
There is already a test below that tests that this works properly

Reply:
I saw, but why not also add some samples to the fixtures?

Reply:
Sure, added